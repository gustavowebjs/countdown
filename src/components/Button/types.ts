export type ButtonSizes = 'small' | 'medium' | 'large';

export type ButtonProps = {
  children: React.ReactNode;
  onClick?: () => void;
  type?: 'button' | 'submit' | 'reset';
  style?: React.CSSProperties;
  className?: string;
  size?: ButtonSizes;
};
