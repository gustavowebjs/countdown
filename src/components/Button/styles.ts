import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const Button = styled.button`
  background: #ff9000;
  height: 42px;
  border-radius: 10px;
  border: 0;
  padding: 0 25px;
  color: #312e38;
  cursor: pointer;

  :hover {
    opacity: 0.9;
  }
`;
