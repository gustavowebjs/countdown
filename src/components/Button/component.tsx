import React, { ReactElement } from 'react';
import * as S from './styles';
import * as T from './types';

const Button = (props: T.ButtonProps): ReactElement => {
  const { children, onClick } = props;

  return (
    <S.Container>
      <S.Button onClick={onClick}>{children}</S.Button>
    </S.Container>
  );
};

export default Button;
