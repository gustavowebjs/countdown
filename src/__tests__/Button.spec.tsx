import { shallow } from 'enzyme';
import sinon from 'sinon';
import Button from '../components/Button';

describe('Button Component', () => {
  it('Should button component render correctly', () => {
    const wrapper = shallow(<Button>Test</Button>);
    expect(wrapper.text()).toEqual('Test');
    expect(wrapper).toMatchSnapshot();
  });

  it('Should button clicks work correctly', () => {
    const onButtonClick = sinon.spy();

    const wrapper = shallow(
      <Button className="button" onClick={onButtonClick}>
        Test
      </Button>
    );

    wrapper.find('.button').simulate('click');
    expect(onButtonClick).toHaveProperty('callCount', 1);
    expect(wrapper).toMatchSnapshot();
  });
});
