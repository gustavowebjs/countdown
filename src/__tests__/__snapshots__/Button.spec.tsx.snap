// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should button clicks work correctly 1`] = `
<styled.div>
  <styled.button
    className="button"
    onClick={[Function]}
    size="medium"
  >
    Test
  </styled.button>
</styled.div>
`;

exports[`Button Component Should button component render correctly 1`] = `
<styled.div>
  <styled.button
    size="medium"
  >
    Test
  </styled.button>
</styled.div>
`;
